### PR TITLE
Blog-1102:Added Signup API and  Login API

### PR DESCRIPTION
**Pull request by jackupsurya**
I've added Signup API and  Login API.

### CODE REVIEW

File: driver/driver.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package driver
```
Comment (Niru2k) [inline driver/driver.go]: Don't commit the empty file.

File: handlers/handlers.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package handlers
+
+import (
+	"blog/helper"
+	"blog/models"
+	"strconv"
+	"time"
+
```
Comment (Niru2k) [inline handlers/handlers.go:16]: Add the comment as Third party package

File: handlers/handlers.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package handlers
+
+import (
+	"blog/helper"
```
Comment (Niru2k) [inline handlers/handlers.go]: Add a comment as user defined package

File: handlers/handlers.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	"blog/models"
+	"strconv"
```
Comment (Niru2k) [inline handlers/handlers.go:15]: Add a comment as inbuild package

File: handlers/handlers.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+// This is for Signup
+func Signup(c *fiber.Ctx) error {
+	var data models.User
+	check := false
+
+	//Get user details from request body
+	if err := c.BodyParser(&data); err != nil {
+		return err
+	}
+	if data.Email == "" || data.Id <= 0 || data.Password == "" || data.Username == "" || data.Role == "" {
```
Comment (Niru2k) [inline handlers/handlers.go]: Need to specify the error, Don't give simply "missing credentials"

File: handlers/handlers.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+		return c.JSON(fiber.Map{
+			"message": "missing credentials",
+		})
+	}
+
+	for _, val := range models.Database {
+		if val.Id == data.Id || val.Email == data.Email {
```
Comment (Niru2k) [inline handlers/handlers.go]: Why if happen when the Id is correct and Email is wrong

File: repository/repository.go [hunk 7]
```diff
@@ -0,0 +1,2 @@
+package repository
```
Comment (Niru2k) [inline repository/repository.go:1]: If you doesn't use the file, Don't commit it.

File: router/router.go [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package router
+
+import (
+	"blog/handlers"
```
Comment (Niru2k) [inline router/router.go:5]: Here also need to add comments to the packages like third party or build-in or user defined

File: router/router.go [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+func Router() {
+	app := fiber.New()
+	app.Post("/signup", handlers.Signup)
+	app.Post("/login", handlers.Login)
+
+	//start a server
+	log.Println("Server starts in port 8000.....")
+	app.Listen("localhost:8000")
```
Comment (Niru2k) [inline router/router.go]: Don't use "localhost:8000" just use ":8000"

File: driver/driver.go (hunk 1 above)
Comment (jackupsurya) [reply driver/driver.go]: Ok

File: handlers/handlers.go (hunk 6 above)
Comment (jackupsurya) [reply handlers/handlers.go]: Here, we have to check only the Email is exist or not. Id validation is not necessary. So I will ignore that id validation.

Comment (Niru2k) [reply handlers/handlers.go]: Then remove it

**Pull request merged**